cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -194,11 +196,20 @@ func Run(c *config.CompletedConfig, stopCh <-chan struct{}) error {
 				// If one isn't, we'll timeout and exit when our client builder is unable to create the tokens.
 				klog.Warningf("--use-service-account-credentials was specified without providing a --service-account-private-key-file")
 			}
-			clientBuilder = controller.SAControllerClientBuilder{
-				ClientConfig:         restclient.AnonymousClientConfig(c.Kubeconfig),
-				CoreClient:           c.Client.CoreV1(),
-				AuthenticationClient: c.Client.AuthenticationV1(),
-				Namespace:            "kube-system",
+			if utilfeature.DefaultFeatureGate.Enabled(features.TokenRequest) {
+				//Dynamic builder will use TokenRequest feature and refresh service account token periodically
```

Comment:
this must detect if the token request API is available and fall back

Reply:
You can do this with the discovery client.

Reply:
@mikedanese  I think I already do this as you said in `shouldTurnOnDynamicClient`